More resource docs cleanup

tas50 opened the pull request:
- Due to a bug in the windows_task resource wasn't showing up with the resource inspector before. This is fixed so now we get to fix some issues.
- Remove some very old Chef 12 "new in" notes that just pollute the docs at this point
- Fix some formatting
- Add more common options
- Fix the italic "new in" notes

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Add missing properties to windows_task resource

Due to a bug in the resource it wasn't showing up with the resource
inspector before. This is fixed so now we get to fix some issues.

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Remove carriage returns from docs

This make our dtags different in hidden ways and it's super hard to
track down without enabling hidden characters

Signed-off-by: Tim Smith <tsmith@chef.io>

kagarmoe commented on chef_master/source/resource_windows_task.rst:
```diff
@@ -1,251 +1,266 @@
+
+   The task description.
+
+   New in Chef Client 14.7.
```
Version annotations should be **bold**

tas50 replied:
Fixed

tas50 committed:
Don't mention very old Chef 12 releases

I left in more recent notes about Chef 12, but these are just ancient and we shouldn't litter our notes with information that isn't relevant anymore.

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Fix new in chef client formatting

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Bring in a few more changes from automation

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Remove extra whitespace

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Reporting -> Automate in the config docs

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Squash a "new in" for Chef 13

Just explain the behavior. If someone is upgrading from Chef 12 this is called out in our docs

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Remove 2 unused configs

knife syntax checking was removed.
yum has been refactored to not honor this config anymore

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
dtag fixes

Signed-off-by: Tim Smith <tsmith@chef.io>

mjingle reviewed (approved):
Thank you for all this work!

mjingle committed:
Merge branch 'master' into resource_cleanup_v5
